scripts/run/gce/logger_daemon.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/python
+
+# Add the following to jbox.user
+# 'cloud_host' : {
+#     ...,
+#     'log-interval': 5, # seconds
+# }
+
+
+from time import sleep, time
+import os
+from sys import argv
+import httplib
+from socket import gethostname
+from oauth2client.client import GoogleCredentials
+from googleapiclient.discovery import build
+import errno
+
+if len(argv) < 2:
+    print('USAGE: ./logger_daemon.py <log_files_paths>')
+    exit(-1)
+
+conf = None
+with open('/jboxengine/conf/jbox.user') as f:
+    conf = eval(f.read())
+LOGPATHS = argv[1:]
+ch = conf['cloud_host']
+INTERVAL = ch['log-interval']
+creds = GoogleCredentials.get_application_default()
+conn = build('logging', 'v2beta1', credentials=creds)
+MYNAME = gethostname()
+logurl = 'projects/%s/logs/%s' % (ch['install_id'], MYNAME)
+
+def retry_on_bsl(f):
+    def func(*args, **kwargs):
+        try:
+            return f(*args, **kwargs)
+        except httplib.BadStatusLine as bsl:  # Retry on bad status line
+            return f(*args, **kwargs)
+        except IOError as e:
+            if e.errno == errno.EPIPE:        # Retry on broken pipe errno 32
+                return f(*args, **kwargs)
+            else:
+                raise
+    return func
+
+@retry_on_bsl
+def cloud_log(line, sev=None):
+    if not sev:
+        try:
+            sev = line.split(' - ')[1]
+            if sev not in ['INFO', 'DEBUG', 'WARNING', 'CRITICAL', 'ERROR']:
+                sev = 'INFO'
+        except:
+            sev = 'INFO'
+
+    ent = conn.entries()
+    body = {
+        'entries': [
+            {
+                'severity': sev,
+                'textPayload': line,
+                'logName': logurl,
+                'resource': {
+                    'labels': {
+                        'instance_id': 'testbox',
+                        'zone': 'us-central1-c'
+                    },
+                    'type': 'gce_instance'
+                }
+            }
+        ]
+    }
+    ent.write(body=body).execute()
+
+def log(filenames):
+    file_handles = [open(f) for f in filenames]
+    while True:
+        start = time()
+        for f in file_handles:
+            for line in f.read().split('\n'):
+                if line:
+                    if f.name.endswith('_err.log'):
+                        cloud_log(line, 'ERROR')
+                    else:
+                        cloud_log(line)
+        sleep_time = INTERVAL - (time() - start)
+        if sleep_time < 0:
+            sleep_time = 0
+        sleep(sleep_time)
+
+if __name__ == '__main__':
+    files = []
+    for l in LOGPATHS:
+        files.extend([os.path.join(l, f) for f in os.listdir(l) if f.endswith('.log')])
+    log(files)
```

Comment:
Since `files` is created only once at the beginning, how does this:
- avoid race condition during startup (e.g. logger daemon may start up before anything else starts up and find no log files)
- discover new log files created later (e.g. error.log may get created later only on an error)
- handle log rotation (when a log file gets rotated, the open file handle will continue referring to the old file which would no longer be appended to)